Upstream Merge #1525 | Station AI Features and Fixes (Also General Fixes)

Fansana opened the pull request:
Upstream PR: https://github.com/simple-station/einstein-engines/pull/1525
<!--
This is a semi-strict format, you can add/remove sections as needed but the order/format should be kept the same
Remove these comments before submitting
-->

# Description

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Check the changelog for the full list.

---

# Changelog

<!--
You can add an author after the `:cl: EinsteinEngines` to change the name that appears in the changelog (ex: `:cl: EinsteinEngines Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl: EinsteinEngines
- add: Added Holopads (unmapped)
- add: Intellicards are now useful for removing/adding a Station AI's brain.
- add: Added the Communications Console to Station AI actions.
- add: AI now has a warp point.
- add: Added more things for the AI to press.
- add: More AI laws have been added.
- fix: Fixed the mail system
- fix: Fixed AI actions
- fix: Fixed invalid spawns for station AI breaking and ruining your ability to play it.
- fix: The Station AI's name will now properly send in "arrived to the station" announcements.
- fix: Changed the CPR sound to simply not loop until fixed.
- fix: Fixed unlocalized messages being sent for the random sentience event.

sleepyyapril committed:
Station AI Features and Fixes (Also General Fixes) (#1525)

<!--
This is a semi-strict format, you can add/remove sections as needed but
the order/format should be kept the same
Remove these comments before submitting
-->

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Check the changelog for the full list.

---

<!--
You can add an author after the `:cl:` to change the name that appears
in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- add: Added Holopads (unmapped)
- add: Intellicards are now useful for removing/adding a Station AI's
brain.
- add: Added the Communications Console to Station AI actions.
- add: AI now has a warp point.
- add: Added more things for the AI to press.
- add: More AI laws have been added.
- fix: Fixed the mail system
- fix: Fixed AI actions
- fix: Fixed invalid spawns for station AI breaking and ruining your
ability to play it.
- fix: The Station AI's name will now properly send in "arrived to the
station" announcements.
- fix: Changed the CPR sound to simply not loop until fixed.
- fix: Fixed unlocalized messages being sent for the random sentience
event.

---------

Co-authored-by: metalgearsloth <31366439+metalgearsloth@users.noreply.github.com>
Co-authored-by: ScarKy0 <106310278+ScarKy0@users.noreply.github.com>
Co-authored-by: Zachary Higgs <compgeek223@gmail.com>
Co-authored-by: MendaxxDev <153332064+MendaxxDev@users.noreply.github.com>
Co-authored-by: chromiumboy <50505512+chromiumboy@users.noreply.github.com>
Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

sleepyyapril committed:
FTL Fixes (#1539)

This does cause it to warp onto the wrong airlock (at arrivals), but
that's a map issue.

(Yes, I tested it)

# Changelog

<!--
You can add an author after the `:cl:` to change the name that appears
in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- fix: Fixes arrivals destroying all nearby doors/walls/etc.

---------

Co-authored-by: Plykiya <58439124+Plykiya@users.noreply.github.com>
Co-authored-by: MilenVolf <63782763+MilenVolf@users.noreply.github.com>
Co-authored-by: Gansu <68031780+GansuLalan@users.noreply.github.com>
Co-authored-by: aa5g21 <aa5g21@soton.ac.uk>
Co-authored-by: metalgearsloth <31366439+metalgearsloth@users.noreply.github.com>
(cherry picked from commit cc858242c5c2d75faa09169f85ca50043dc4ea7c)

Commit:
fix function parameter

ShadesMars commented:
Without looking directly theres missing textures files its seems.

Memeji reviewed (dismissed):
Issues will need to be resolved before testing.

M3739 commented:
I shall attempt to take a crack at this later on today.

M3739 committed:
Merge branch 'Gnosis' into upstream/1525

# Conflicts:
#	Content.Server/Medical/CPR/CPRSystem.cs

M3739 committed:
Fix Ai Accessible Consoles (#1485)

# Description

This fixes the bug where consoles that are whitelisted by AI can only be
accessed by AI. It does so by cherrypicking
https://github.com/space-wizards/space-station-14/pull/32012
and https://github.com/space-wizards/space-station-14/pull/31170 (which
the first cherrypick requires)
and now https://github.com/space-wizards/space-station-14/pull/29094

Also fixes AI Speech/Languages issue

DO NOT SQUASH THIS, IT'S THREE PRS.

(cherry picked from commit 81bab8abe06ae96f58fd5e7048c6e7f4218a8654)

# Conflicts:
#	Resources/Audio/Machines/attributions.yml
#	Resources/Prototypes/Entities/Mobs/Player/silicon.yml
#	Resources/Textures/Interface/Actions/actions_ai.rsi/meta.json

M3739 committed:
Station AI Features and Fixes (Also General Fixes) (#1525) (Again)

<!--
This is a semi-strict format, you can add/remove sections as needed but
the order/format should be kept the same
Remove these comments before submitting
-->

# Description

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Check the changelog for the full list.

---

# Changelog

<!--
You can add an author after the `:cl:` to change the name that appears
in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- add: Added Holopads (unmapped)
- add: Intellicards are now useful for removing/adding a Station AI's
brain.
- add: Added the Communications Console to Station AI actions.
- add: AI now has a warp point.
- add: Added more things for the AI to press.
- add: More AI laws have been added.
- fix: Fixed the mail system
- fix: Fixed AI actions
- fix: Fixed invalid spawns for station AI breaking and ruining your
ability to play it.
- fix: The Station AI's name will now properly send in "arrived to the
station" announcements.
- fix: Changed the CPR sound to simply not loop until fixed.
- fix: Fixed unlocalized messages being sent for the random sentience
event.

---------

Co-authored-by: metalgearsloth <31366439+metalgearsloth@users.noreply.github.com>
Co-authored-by: ScarKy0 <106310278+ScarKy0@users.noreply.github.com>
Co-authored-by: Zachary Higgs <compgeek223@gmail.com>
Co-authored-by: MendaxxDev <153332064+MendaxxDev@users.noreply.github.com>
Co-authored-by: chromiumboy <50505512+chromiumboy@users.noreply.github.com>
Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>
(cherry picked from commit 3e8a7d9b00e19e160321eb81d69a884189dfa4e6)

# Conflicts:
#	Content.Server/Mail/Systems/MailSystem.cs
#	Content.Server/Medical/CPR/CPRSystem.cs
#	Content.Server/StationEvents/Events/RandomSentienceRule.cs
#	Resources/Locale/en-US/interaction/interaction-popup-component.ftl
#	Resources/Prototypes/Entities/Mobs/Player/silicon.yml
#	Resources/Textures/Interface/Actions/actions_ai.rsi/meta.json

M3739 committed:
fix function parameter

(cherry picked from commit bed104ed49100d1ab7ae206f8a158b16eab06fb5)

# Conflicts:
#	Content.Server/Mail/Systems/MailSystem.cs

M3739 commented:
Finally managed to touch up on this today. Found out the missing PR we needed: https://github.com/Simple-Station/Einstein-Engines/pull/1485

So, I cherry-picked that, resolved the merge conflict, applied 1525 again, then reapplied Fansana's function fix by cherry-picking it.

Local YAML linter test returned no errors as of writing. For now though, I gotta do some IRL things. But this should *hopefully* resolve the test errors.

M3739 committed:
*sigh...*

M3739 commented:
tfw you forget to commit and push your local changes when writing your comment

M3739 committed:
Remove extra `deviceNetId`  from `emergency_meta.yml`

M3739 commented:
Welp. Checks do pass. And it does launch.
I've done my part, t'would seem

ghost commented:
@M3739 uh could you do a review so I can merge? Also thanks a lot.

M3739 reviewed (commented):
![image](https://github.com/user-attachments/assets/b120cd8e-edc8-4af2-84f2-2e1d9500dcc2)
![image](https://github.com/user-attachments/assets/ea6aa865-1774-4c7b-9180-ee7e71e1a3a0)
![image](https://github.com/user-attachments/assets/7cbe24f6-d0fa-413d-ac4e-490bf4098d18)
![image](https://github.com/user-attachments/assets/d487a5e5-e419-45ae-a3c9-a8b5ac71b511)
Personally, I am unsure if it's *supposed* to be where my "eye" is the hologram, not the AI bracket thingy majig. It does still hide, though.
![image](https://github.com/user-attachments/assets/80594fe9-75a4-4bef-96ae-04ed183a6b34)
![image](https://github.com/user-attachments/assets/0bd20425-ce2e-4606-aab3-cf447625b617)
![image](https://github.com/user-attachments/assets/dfc2ede7-467a-4ffa-b7fc-1c6b79ae89b4)
![image](https://github.com/user-attachments/assets/0eedae9f-0de7-4652-92d8-5b8466e0c908)

Overall, seems to work. Just need to check one thing in the code, and enable something related to polymorphing and intellicards.

M3739 reviewed (approved):
Everything seems to check out on my end.
As it turns out, the Intellicard stuff has yet to be `shared` so as a result, DV 3539 cannot be applied proper. Yet.

M3739 commented:
Alright, so the eye will come later down the road, as in EE 1680.

I'd say it's ready, @Fansana.

Memeji commented:
Going to do some further little testing just to make sure.
Gonna try and get this in for the next release.

Memeji commented:
There's an issue with the arrivals shuttle deleting walls when docking with the station.
Can see this by running in release and going through the arrivals shuttle to get to the station.

Memeji reviewed (changes requested):
Need to fix the issue with arrivals destroying walls when arriving to the station.

M3739 commented:
Alright. EE 1539 directly addresses this issue. Applied.

Memeji commented:
![image](https://github.com/user-attachments/assets/d2e05669-3b24-467b-8dde-a41fc6fbfb3d)

Still occurs at arrivals it seems.

Memeji reviewed (approved):
Fix did work, didn't apply to my instance but testing on others shown it worked.
